hydra-node/src/Hydra/Chain/Direct/State.hs

```diff
@@ -587,29 +526,23 @@ observeFanout st tx = do
  where
   ClosedState{ctx} = st
 
-instance ObserveTx ClosedState IdleState where
-  observeTx = observeFanout
-
 -- * Observe any transition
 
 -- | Observe a transition without knowing the starting or ending state. This
--- function does enumerate and try all 'transitions' of some given
--- 'SomeOnChainHeadState'. To do that, this function uses the 'HasTransitions'
--- and 'ObserveTx' type classes.
-observeSomeTx ::
-  Tx ->
-  SomeOnChainHeadState ->
-  Maybe (OnChainTx Tx, SomeOnChainHeadState)
-observeSomeTx tx (SomeOnChainHeadState (st :: st)) =
-  asum $ (\(TransitionTo _ p) -> observeSome p) <$> transitions @st
- where
-  observeSome ::
-    forall st'.
-    (ObserveTx st st', Typeable st', Eq st', Show st', HasTransitions st', HasKnownUTxO st') =>
-    Proxy st' ->
-    Maybe (OnChainTx Tx, SomeOnChainHeadState)
-  observeSome _ =
-    second SomeOnChainHeadState <$> observeTx @st @st' st tx
+-- function should try to observe all relevant transitions given some
+-- 'ChainState'.
+observeAllTx :: Tx -> ChainState -> Maybe (OnChainTx Tx, ChainState)
```

Comment:
Suggestion: Put `observeAllTx` before the individual `observeXX`. This helps understand the context in which each of these function is used and provide the "state machine" structure before its implementation